### PR TITLE
Add support for watch.

### LINKED ISSUE #4: Support Watch
Watch is a non-standard method that swagger-codegen does not support and we need to add support for it like other clients (e.g. python [here](https://github.com/kubernetes-client/python-base/tree/a41c44715241552de73361673152f3f0d0bb9bc4/watch))

### PR DESCRIPTION
**Pull request by brendandburns**
Closes #4 

@drubin 

Thanks!

### CODE REVIEW

File: kubernetes/src/kubernetes/watch.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            @client = client
+        end
+        
+        def connect(path, &block)
+            opts = {
+                :auth_names => ['BearerToken']
+            }
+            request = @client.build_request('GET', path + '?watch=true', opts)
+            request.on_body do |chunk|
+                parts = chunk.split(/\n/)
```
Comment (cben) [inline kubernetes/src/kubernetes/watch.rb]: I'm not sure transport chunks always end in a newline.  Last part might be incomplete, may need to buffer it and concat with first part of next chunk?

Comment (brendandburns) [reply kubernetes/src/kubernetes/watch.rb]: Good point. Fixed (and added some tests to validate)

**Comment by brendandburns** [comment]
@cben thanks for the feedback, comment addressed!

**Comment by drubin** [comment]
Feels like we should rebase the `Add support for Azure AAD login. Fix some bugs.` out of this PR though?

File: kubernetes/src/kubernetes/watch.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        pieces.each do |part|
+          yield JSON.parse(part)
+        end
+      end
+      request.run
+    end
+
+    def split_lines(last, chunk)
+      data = chunk
+      data = last + "\n" + data unless last.empty?
```
Comment (cben) [inline kubernetes/src/kubernetes/watch.rb]: the idea was that a single line could(?) be split between prev chunk and this one, so I think this should just be
```suggestion
data = last + chunk
```

Comment (brendandburns) [reply kubernetes/src/kubernetes/watch.rb]: No, because if last itself is empty you don't want to randomly prepend a newline...  (I had that bug)

Comment (cben) [reply kubernetes/src/kubernetes/watch.rb]: Right but I'm saying you don't want the newline between them!
Say first chunk is cut `{"Foo": 0}\n{"ba`
and next chunk continues it `r": 1}\n{"baz`.
If you join with a newline, the 2nd json will not parse properly.


Then, prepending an empty string is harmless, you can keep the `unless last.empty?` or drop it, as you prefer.

Comment (brendandburns) [reply kubernetes/src/kubernetes/watch.rb]: fixed.

**Comment by brendandburns** [comment]
@cben comment addressed

@drubin let's merge #39 and then I will rebase to remove that commit.

Thanks!

**Comment by brendandburns** [comment]
@drubin @cben rebased and ready to merge.

**Comment by brendandburns** [comment]
@drubin @cben ok, this is actually really ready for re-review :)

**Comment by cben** [comment]
Do kube_condig.rb changes belong in this PR?  No opinion on those.
The rest LGTM :100:

**Comment by brendandburns** [comment]
@cben the same fix is in #44 so let's merge that first and I'll rebase?

**Comment by cben** [comment]
Just to be clear, I'm not a merger/maintainer here.
I'm following this repo to "spy for the opposition" :wink:  and got interested in this one...

**Comment by drubin** [comment]
@brendandburns mind rebasing/fixing/closing depending on how you see fit?

**Comment by brendandburns** [comment]
@drubin rebased. please take another look.

Thanks!

**Pull request merged**